### PR TITLE
[REEF-1716] Implement HelloREEF example for YARN Unmanaged AM mode

### PR DESCRIPTION
**Pull request by motus**
**NOTE:** Merge this PR **after** #1248 

JIRA: [REEF-1716](https://issues.apache.org/jira/browse/REEF-1716)

Closes #

### CODE REVIEW

**Comment by motus** [comment]
**NOTE:** CI builds currently fail because #1248 has to be merged before this one. Please try it *after* merging #1248 on Hadoop 2.7.3+ (it hangs on Hadoop 2.7.1 that we have on CISL cluster due to the `NullPointerException` in YARN - see [YARN-4452](https://issues.apache.org/jira/browse/YARN-4452))

**Comment by motus** [comment]
Will do. We kind of have it in [REEF-1710](https://issues.apache.org/jira/browse/REEF-1710) (PR #1224), but I'll add a REEF-based test in a subsequent PR

**Comment by jwang98052** [comment]
Do you have instruction for how to run it on YARN?

File: lang/java/reef-examples/src/main/java/org/apache/reef/examples/hello/HelloREEFYarnUnmanagedAm.java [hunk 1]
```diff
@@ -0,0 +1,89 @@
...
+import org.apache.reef.util.EnvironmentUtils;
+import org.apache.reef.util.ThreadLogger;
+
+import java.util.logging.Level;
+import java.util.logging.Logger;
+
+/**
+ * The Client for running HelloREEF on YARN in unmanaged AM mode.
+ */
+public final class HelloREEFYarnUnmanagedAm {
```
Comment (markusweimer) [inline lang/java/reef-examples/src/main/java/org/apache/reef/examples/hello/HelloREEFYarnUnmanagedAm.java]: QQ: Shall we use `Am` or `AM` here as the name? After all, it is an acronym.

**Comment by jwang98052** [comment]
Do you have local RunTime version for it?

**Comment by jwang98052** [comment]
Since #1248 has been merged, you might want to trigger another Travis CI.

File: lang/java/reef-examples/src/main/java/org/apache/reef/examples/hello/HelloREEFYarnUnmanagedAm.java (hunk 1 above)
Comment (motus) [inline lang/java/reef-examples/src/main/java/org/apache/reef/examples/hello/HelloREEFYarnUnmanagedAm.java]: Sure, I'll fix that. This commit should also trigger a new build on CI server.

**Comment by motus** [comment]
@jwang98052 Unmanaged AM mode is YARN-specific - this is why we have a dedicated runtime configuration for it. For the local runtime, though, we already have some unit tests, e.g. `REEFEnvironmentDriverTest` that runs the driver in the client process.

**Comment by jwang98052** [comment]
There is one test failure from Trvis-CI
Failed tests: 
  RemoteTest.testRemoteTest:180 testRemoteTest takes too long and times out : 0 out of 6 events

but I don't think it is related. I will run entire test locally again before the merge.

**Pull request closed without merging**